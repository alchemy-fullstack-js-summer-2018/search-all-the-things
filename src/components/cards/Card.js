import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Card extends Component {

  static propTypes = {
    card: PropTypes.object.isRequired
  };

  render() {
    const { card } = this.props;
    const { imageUrl, name } = card;

    return (
      <li>
        <img src={imageUrl} alt={name}/>
      </li>
    );
  }
}