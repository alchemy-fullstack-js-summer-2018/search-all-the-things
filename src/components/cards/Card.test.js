import React from 'react';
import  { shallow } from 'enzyme';
import Card from './Card';
import toJSON from  'enzyme-to-json';

const card = {
  'source': {
    'id': null,
    'name': 'magic.com'
  },
  'imageUrl': 'image.jpg',
  'name': 'Adorable Kitten',
  'setName': 'set name'
};

describe('Card', () => {

  it('renders card correctly', () => {
    const wrapper = shallow(<Card card={card}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});