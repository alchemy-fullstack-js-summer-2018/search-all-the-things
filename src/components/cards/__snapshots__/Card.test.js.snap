// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders card correctly 1`] = `
<li>
  <Link
    replace={false}
    to="/detail/undefined"
  >
    <img
      alt="Adorable Kitten(set name)"
      src="image.jpg"
    />
  </Link>
</li>
`;
